lib/api/src/lib.rs

```diff
@@ -118,3 +120,118 @@ pub use wasmer_engine_native::{Native, NativeArtifact, NativeEngine};
 
 /// Version number of this crate.
 pub const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+// --------------------------------------------------------------
+// TODO: put this in a proper location, just prototyping for now:
+// TODO: rename everything, all names are throw-away names
+
+/// Prototype trait for finishing envs.
+pub trait WasmerEnv {
+    /// The function that Wasmer will call on your type to let it finish
+    /// instantiating.
```

Comment:
We must explain the workflow here. When the “instantation finishes”? Is it really after the Wasmer instantiation, or before something else? What actions the user can do? What can the user change? An example of a `finish` function would be helpful, at least to explain why it is required or necessary?